flexport/latitude
# Prevent header from remounting on table rerender

This is the beginning of the table refactor to prevent cells from remounting on table rerender. This PR moves the `innerElementType` function definition out of the table render to prevent the header row from remounting on every table re-render. The react-window [docs](https://github.com/bvaughn/react-window) link to a [sticky header example](https://codesandbox.io/s/0mk3qwpl4l) and it uses React context to pass information to the innerElementType.

Manual test plan:
Run storybook and go to the Rerender Table Test.
Add `React.useEffect(() => console.log('header row mount'), [])` to the `HeaderRow` function and see that it does not remount on table re-render.
